.github/workflows/ci.yml

```diff
@@ -176,16 +184,32 @@ jobs:
         run: source ci/github/install.sh
 
       - name: Setup coverage collection
-        if: matrix.coverage
+        if: "matrix.coverage && !matrix.coverity"
         run: ci/github/codecov.sh "setup"
 
       - name: Run tests
+        if: "!matrix.coverity"
         run: ci/build.sh
 
       - name: Upload coverage
-        if: matrix.coverage
+        if: "matrix.coverage && !matrix.coverity"
         run: ci/codecov.sh "upload"
 
+      - name: Setup coverity
```

Comment:
Can't we combine those 2 steps? Basically do the check in `ci/github/coverity.sh` and print an error if COVERITY_SCAN_TOKEN is empty? If a library author doesn't want it, he can remove the config line. Same as with coverage

Reply:
Sure.

Reply:
I would rather have the coverity step not activate if the github secrets are not present, so it shows up as a job that was skipped in CI instead of one that ran, so I think we should keep this.  I can change the name to "Check for coverity secrets" I suppose.

Reply:
Why would you want to run the whole process and then abort/skip the coverity step here? I mean you either have a job with "coverity:yes" in the matrix or you don't. I don't see a purpose of setting up an environment and then skipping the "build" step

Reply:
Fair enough I will do that.  If the matrix entry is enabled and the secrets are not set then it is an error.